pmlc/dialect/stdx/transforms/boundscheck.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/ADT/TypeSwitch.h"
+#include "mlir/Conversion/AffineToStandard/AffineToStandard.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/DebugStringHelper.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "pmlc/dialect/stdx/transforms/boundscheck.h"
+#include "pmlc/util/logging.h"
+
+using namespace mlir; // NOLINT
+
+namespace pmlc::dialect::stdx {
+
+namespace {
+template <typename LoadStoreOp>
+class BoundsCheckGenerator {
+  LoadStoreOp op;
+  Location loc;
+  ModuleOp module;
+  Type i32Type;
+  Type indexType;
+
+private:
+  BoundsCheckGenerator(LoadStoreOp op, Builder builder)
+      : op(op), loc(op.getLoc()),
+        module(op.template getParentOfType<ModuleOp>()),
+        i32Type(builder.getIntegerType(32)), indexType(builder.getIndexType()) {
```

Comment:
Minor point, no need to construct a builder just for this:  IntegerType::get(32, op.getContext()) also works.